OpenRA.Mods.Common/Traits/Render/ProductionBar.cs

```diff
@@ -40,34 +41,44 @@ public ProductionBar(Actor self, ProductionBarInfo info)
 			this.info = info;
 		}
 
+		void INotifyCreated.Created(Actor self)
+		{
+			produces = self.TraitsImplementing<Production>().ToArray();
+			FindQueue();
+		}
+
 		void FindQueue()
 		{
-			var type = info.ProductionType ?? self.Info.TraitInfo<ProductionInfo>().Produces.First();
+			// Use all queues when no specific one is given
+			if (info.ProductionType == null)
+				return;
 
 			// Per-actor queue
 			// Note: this includes disabled queues, as each bar must bind to exactly one queue.
 			queue = self.TraitsImplementing<ProductionQueue>()
-				.FirstOrDefault(q => type == null || type == q.Info.Type);
+				.FirstOrDefault(q => info.ProductionType == q.Info.Type);
 
 			if (queue == null)
 			{
 				// No queues available - check for classic production queues
 				queue = self.Owner.PlayerActor.TraitsImplementing<ProductionQueue>()
-					.FirstOrDefault(q => type == null || type == q.Info.Type);
+					.FirstOrDefault(q => info.ProductionType == q.Info.Type);
 			}
-
-			if (queue == null)
-				throw new InvalidOperationException("No queues available for production type '{0}'".F(type));
-		}
-
-		void INotifyCreated.Created(Actor self)
-		{
-			FindQueue();
 		}
 
 		void ITick.Tick(Actor self)
 		{
-			var current = queue.CurrentItem();
+			var q = queue;
```

Comment:
Why do we need this variable?

Reply:
How else do you suggest I handle the case when `queue == null` without reassigning the `queue` variable? (We could e.g. move the `value` assignment in if statements, I thought this is the cleaner approach though.)